### ISSUE TITLE
cursor moves to wrong position after formatting 

### ISSUE THREAD

**Issue by paulvanbrenk** [microsoft/TypeScript#15113]
Copied from https://github.com/Microsoft/nodejstools/issues/49

`var result = {'count': 0`<cursor>`}`

press enter

``` javascript
var result = {
        'count': 0

   <cursor> }
```

(brace completion turned on, only seems to repro while typing it out

**Comment by aozgaa** [comment]
To expand, given
``` ts
var result = {'count': 0^}
```
with cursor indicate by `^`, pressing `Enter` triggers the issue.

## Expected Behavior

``` ts
var result = {
    'count': 0
^}
```

## Actual Behavior

``` ts
var result = {
    ^'count': 0
}
```

Appears to be VS-specific, as activating `editor.formatOnType` in vscode doesn't trigger the error.
